Using android plugin with godot 4 is crashing the game (No implementation found for void GodotPlugin.nativeRegisterSingleton)

### Godot version

4.0 alpha

### System information

android 11

### Issue description

Custom build is working fine without any crash but using any android plugin is making the crash( not even show any launch screen)
It works fine in godot  3.4
``` java.lang.UnsatisfiedLinkError: No implementation found for void org.godotengine.godot.plugin.GodotPlugin.nativeRegisterSingleton(java.lang.String, java.lang.Object) (tried Java_org_godotengine_godot_plugin_GodotPlugin_nativeRegisterSingleton and Java_org_godotengine_godot_plugin_GodotPlugin_nativeRegisterSingleton__Ljava_lang_String_2Ljava_lang_Object_2) ```

### Steps to reproduce

use custom build with any plugin like godot-share

### Minimal reproduction project

_No response_

No crash, the plugin is working fine too.

Some headups, will create a separate issue if required.
- target and compiled sdk version are now required to be 31 instead of 30
- Installing through USB debugging is throwing error but exporting apk is fine
```Task :compileDebugJavaWithJavac FAILED
D:\demo\android\build\src\com\godot\game\GodotApp.java:33: error: package org.godotengine.godot does not exist
import org.godotengine.godot.FullScreenGodotApp;
                            ^
D:\demo\android\build\src\com\godot\game\GodotApp.java:41: error: cannot find symbol
public class GodotApp extends FullScreenGodotApp {
                              ^
  symbol: class FullScreenGodotApp
D:\demo\android\build\src\com\godot\game\GodotApp.java:42: error: method does not override or implement a method from a supertype
	@Override
	^
D:\demo\android\build\src\com\godot\game\GodotApp.java:44: error: cannot find symbol
		setTheme(R.style.GodotAppMainTheme);
		^
  symbol:   method setTheme(int)
  location: class GodotApp
D:\demo\android\build\src\com\godot\game\GodotApp.java:45: error: cannot find symbol
		super.onCreate(savedInstanceState);
		^
  symbol:   variable super
  location: class GodotApp
5 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileDebugJavaWithJavac'.
> Compilation failed; see the compiler error output for detail```

--- a/platform/android/SCsub
+++ b/platform/android/SCsub
@@ -18,6 +18,7 @@ android_files = [
     "jni_utils.cpp",
     "android_keys_utils.cpp",
     "display_server_android.cpp",
+    "plugin/godot_plugin_jni.cpp",
     "vulkan/vulkan_context_android.cpp",
 ]
 

--- a/platform/android/plugin/godot_plugin_jni.cpp
+++ b/platform/android/plugin/godot_plugin_jni.cpp
@@ -126,7 +126,7 @@ JNIEXPORT void JNICALL Java_org_godotengine_godot_plugin_GodotPlugin_nativeEmitS
 		env->DeleteLocalRef(j_param);
 	};
 
-	singleton->emit_signal(SNAME(signal_name), args, count);
+	singleton->emit_signal(StringName(signal_name), args, count);
 }
 
 JNIEXPORT void JNICALL Java_org_godotengine_godot_plugin_GodotPlugin_nativeRegisterGDNativeLibraries(JNIEnv *env, jclass clazz, jobjectArray gdnlib_paths) {

